src/coreclr/System.Private.CoreLib/src/System/RuntimeType.CoreCLR.cs

```diff
@@ -858,35 +849,34 @@ private RuntimeFieldInfo[] PopulateFields(Filter filter)
 
                 private unsafe void PopulateRtFields(Filter filter, RuntimeType declaringType, ref ListBuilder<RuntimeFieldInfo> list)
                 {
-                    IntPtr* pResult = stackalloc IntPtr[64];
-                    int count = 64;
+                    Span<IntPtr> result = stackalloc IntPtr[64];
+                    int count = 0;
 
-                    if (!RuntimeTypeHandle.GetFields(declaringType, pResult, &count))
+                    if (!RuntimeTypeHandle.GetFields(declaringType, ref result, ref count))
                     {
-                        fixed (IntPtr* pBigResult = new IntPtr[count])
-                        {
-                            RuntimeTypeHandle.GetFields(declaringType, pBigResult, &count);
-                            PopulateRtFields(filter, pBigResult, count, declaringType, ref list);
-                        }
+                        result = new IntPtr[count];
+                        bool success = RuntimeTypeHandle.GetFields(declaringType, ref result, ref count);
+                        Debug.Assert(success && result.Length == count);
```

Comment:
This is not guaranteed to succeed. It is possible to add a new field via EnC between the previous GetFields call and this call. It would be more robust to make this a loop that keeps looping until we allocate big enough array. Something like:

```
Span<IntPtr> result = stackalloc IntPtr[64];
while (!RuntimeTypeHandle.GetFields(declaringType, result, out count))
{
    Debug.Assert(count > result.Length);
    result = new IntPtr[count];
}
PopulateRtFields(filter, result.Slice(0, count), declaringType, ref list);
```